lib/src/main/kotlin/com/lemonappdev/konsist/core/declaration/KoDoc.kt

```diff
@@ -0,0 +1,178 @@
+package com.lemonappdev.konsist.core.declaration
+
+import org.jetbrains.kotlin.kdoc.psi.api.KDocElement
+import org.jetbrains.kotlin.kdoc.psi.impl.KDocSection
+
+class KoDoc(private val kDocElement: KDocElement) {
```

Comment:
here is a common "kt type" between `KDocElement` (used in `KoDoc`) and `KtElement` (used in `KoBaseDeclaration`)  - `PsiElement`:

<img width="1282" alt="image" src="https://user-images.githubusercontent.com/530122/234563349-093a7509-9f75-445c-bb4e-18c12d44f933.png">


1. Create `KoPsiDeclaration` class that has one parameter `psiElement` - this will be common class for all of the declarations
2. `KoBaseDeclaration` extends `KoPsiDeclaration` (these declaration names will have to be unified)
4. Move every property that can be moved from  `KoBaseDeclaration` to `KoPsiDeclaration` 
5. Move related extensions as well